### PR TITLE
Val exotica

### PR DESCRIPTION
**Pull request by yimingyanged**
TableDemo
－ added planReachToTableExotica in table demo 
－ Different goal frames for different scenes

ikPlanner
－ Options to run IK and Trajectory planning separately in Matlab and EXOTica

### CODE REVIEW

File: src/python/ddapp/ikplanner.py [hunk 1]
```diff
@@ -219,6 +220,7 @@ def __init__(self, ikServer, robotModel, jointController, handModels):
         self.additionalTimeSamples = 0
         self.useQuasiStaticConstraint = True
         self.pushToMatlab = True
+	self.pushToMatlabIK = True
```
Comment (patmarion) [inline src/python/ddapp/ikplanner.py]: i think these flag could be renamed.  you've introduced a second flag and now it's not too clear what the intention is.

File: src/python/ddapp/ikplanner.py [hunk 2]
```diff
@@ -65,8 +65,9 @@ def runIk(self):
         if nominalPoseName == 'q_start':
             nominalPoseName = self.startPoseName
 
-        if (self.ikPlanner.pushToMatlab is False):
+        if (self.ikPlanner.pushToMatlab is False and self.ikPlanner.pushToMatlabIK is False):
```
Comment (patmarion) [inline src/python/ddapp/ikplanner.py]: using "is False" is a little strange, try using "not" instead

File: src/python/ddapp/ikplanner.py [hunk 3]
```diff
@@ -986,9 +988,10 @@ def planGraspOrbitReachPlan(self, startPose, side, graspFrame, constraints=None,
     def addPose(self, pose, poseName):
         self.jointController.addPose(poseName, pose)
 
-        if self.pushToMatlab:
+        if (self.pushToMatlab is True or self.pushToMatlabIK is True):
             self.ikServer.sendPoseToServer(pose, poseName)
-        else:
+        #else:
+	if (self.pushToMatlab is False or self.pushToMatlabIK is False):
```
Comment (patmarion) [inline src/python/ddapp/ikplanner.py]: seems that either flag being false implies publishing to an external planner.  perhaps the flag could be renamed to make it more clear that it is used to decide when you publish to an external plan.

File: src/python/ddapp/ikplanner.py [hunk 4]
```diff
@@ -65,8 +65,9 @@ def runIk(self):
         if nominalPoseName == 'q_start':
             nominalPoseName = self.startPoseName
 
-        if (self.ikPlanner.pushToMatlab is False):
+        if (self.ikPlanner.pushToMatlab is False and self.ikPlanner.pushToMatlabIK is False):
             self.endPose, self.info = self.ikPlanner.plannerPub.processIK(self.constraints, nominalPoseName=nominalPoseName, seedPoseName=seedPoseName)
+	    print 'Using external (EXOTica) IK planner'
```
Comment (patmarion) [inline src/python/ddapp/ikplanner.py]: remove print statement

File: src/python/ddapp/ikplanner.py (hunk 1 above)
Comment (yimingyanged) [inline src/python/ddapp/ikplanner.py]: The flags intents to separate the solvers for IK pose and IK trajectory, I pushed a new commit with new flag names.

File: src/python/ddapp/startup.py [hunk 5]
```diff
@@ -1044,7 +1044,8 @@ def stop(self):
 
 if 'exo' in drcargs.args():
     if (drcargs.args().exo):
-        ikPlanner.pushToMatlab = False
+        ikPlanner.pushToMatlabIKPose = False
+	ikPlanner.pushToMatlabIKTraj = False
```
Comment (iamwolf) [inline src/python/ddapp/startup.py]: Indentation issue?

**Comment by patmarion** [comment]
I'm not sure if this pull request is still active.  Closing for now.  Please feel free to re-open.

**Pull request closed without merging**